pilot/pkg/security/authz/model/generator.go

```diff
@@ -399,6 +399,17 @@ func (g pathGenerator) permission(key, value string, forTCP bool) (*rbacpb.Permi
 		return nil, fmt.Errorf("%q is HTTP only", key)
 	}
 
+	// TODO(jaellio): Should we interpret the path as a PathMatcher if it is an invalid or unsupported by Istio PathTemplate?
+	// Or should we error?
+	// Ex: "/foo/{*" is not a valid PathTemplate.
+	// Ex: "/foo/{bar}" is an unsupported PathTemplate.
+	//
+	// Should we disallow the use of `*` and `**` in the path if it is a PathTemplate (the path contains `{*}` or `{**}`, and `*` or `**`)?
```

Comment:
I'd be inclined towards disallowing it. I think the chances of someone trying to match a path that genuinely has a `*` in it are vanishingly small, and if they do exist they will raise a feature request to have the validation relaxed.

Reply:
@michaelbannister added logic to disallow the use of `*` and `**` with path templating. Did not add the logic to check for invalid path templates (i.e. `/foo/{*`).